src/main/kotlin/org/rust/ide/annotator/RustHighlightingAnnotator.kt

```diff
@@ -12,7 +12,11 @@ import org.rust.lang.core.psi.util.elementType
 
 // Highlighting logic here should be kept in sync with tags in RustColorSettingsPage
 class RustHighlightingAnnotator : Annotator {
-    override fun annotate(element: PsiElement, holder: AnnotationHolder) = element.accept(object : RustElementVisitor() {
+
+    override fun annotate(element: PsiElement, _holder: AnnotationHolder) {
+        element.accept(object : RustElementVisitor() {
+        var holder = wrap(_holder)
```

Comment:
I think that instead of creating `Highlighter` class we can more naturally declare a couple of local extension functions:

```Kotlin
    override fun annotate(element: PsiElement, holder: AnnotationHolder) {
        fun PsiElement?.highlight(textAttributesKey: TextAttributesKey?) {
            if (this != null && textAttributesKey != null) {
                holder.createInfoAnnotation(element, null).textAttributes = textAttributesKey
            }
        }

        fun PsiElement?.highlight(color: RustColor?) {
            highlight(color?.textAttributesKey)
        }


        element.accept(object : RustElementVisitor() {

            override fun visitAttr(o: RustAttrElement) {
                o.highlight(RustColor.ATTRIBUTE)
            }
```